src/vs/vscode.d.ts

```diff
@@ -2719,6 +2719,57 @@ declare module 'vscode' {
 	}
 
 	/**
+	 * Contains additional information about the context in which
+	 * [completion provider](#CompletionItemProvider.provideCompletionItems) is triggered.
+	 */
+	export interface CompletionContext {
+		/**
+		 * How the completion was triggered.
+		 */
+		readonly trigger: 'auto' | 'manual';
```

Comment:
We don't do string-or-types but (numeric) enums in the API. We should have `CompletionTriggerKind` as in [Roslyn](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionTriggerKind.cs,d64cce26e8f6eaa3) but with members `Invoke`, `TriggerCharacter`, and `Typing?` (and for the future something like `Refine`)